.pipelines/ios_packaging.yml

```diff
@@ -0,0 +1,92 @@
+# packaging pipeline for iOS CocoaPods package
+
+parameters:
+- name: IsReleaseBuild
+  displayName: "Is this is a release build?"
+  type: boolean
+  default: false
+
+jobs:
+- job: IosPackaging
+  displayName: "iOS Packaging"
+
+  pool:
+    vmImage: "macOS-12"
+
+  timeoutInMinutes: 120
+
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: "3.9"
+      addToPath: true
+      architecture: "x64"
+
+  # iOS xcframework build doesn't work with CMake 3.25.1, pin to 3.25.0
+  - script: |
+      python -m pip install cmake==3.25.0
+    displayName: "Install CMake 3.25.0"
+
+  - script: |
+      python ./tools/gen_selectedops.py ./tools/ios/package_ops.config
+    displayName: "Generate selected ops CMake file"
+
+  - script: |
+      python ./tools/ios/build_xcframework.py \
+        --output-dir $(Build.BinariesDirectory)/xcframework_out \
+        --config Release \
+        --cmake-extra-define OCOS_ENABLE_SELECTED_OPLIST=ON
+    displayName: "Build xcframework"
+
+  # <repo root>/.version.txt is generated by a build, so we put the step that reads it after the build step
+  - template: templates/set-package-version-variable-step.yml
+    parameters:
+      IsReleaseBuild: ${{ parameters.IsReleaseBuild }}
+      PackageVersionVariableName: ORT_EXTENSIONS_POD_VERSION
+
+  - script: |
+      python ./tools/ios/assemble_pod_package.py \
+        --staging-dir $(Build.BinariesDirectory)/pod_staging \
+        --xcframework-output-dir $(Build.BinariesDirectory)/xcframework_out \
+        --pod-version ${ORT_EXTENSIONS_POD_VERSION}
+    displayName: "Assemble pod"
+
+  - script: |
+      pod lib lint
+    displayName: "Lint pod"
+    workingDirectory: $(Build.BinariesDirectory)/pod_staging
+
+  - script: |
+      ORT_EXTENSIONS_LOCAL_POD_PATH=$(Build.BinariesDirectory)/pod_staging \
+        pod install
+    displayName: "Install pods for OrtExtensionsUsage"
+    workingDirectory: $(Build.SourcesDirectory)/test/ios/OrtExtensionsUsage
+
+  - template: templates/xcode-build-and-test-step.yml
+    parameters:
+      xcWorkspacePath: '$(Build.SourcesDirectory)/test/ios/OrtExtensionsUsage/OrtExtensionsUsage.xcworkspace'
+      scheme: 'OrtExtensionsUsage'
+
+  # TODO run test on app center
```

Comment:
note: UI test works locally but running into an app center error. might defer this TODO to another PR